actionpack/lib/action_controller/test_case.rb

```diff
@@ -176,12 +176,12 @@ class LiveTestResponse < Live::Response
 
   # Methods #destroy and #load! are overridden to avoid calling methods on the
   # @store object, which does not exist for the TestSession class.
-  class TestSession < Rack::Session::Abstract::SessionHash #:nodoc:
+  class TestSession < Rack::Session::Abstract::PersistedSecure::SecureSessionHash #:nodoc:
```

Comment:
I am not sure why the superclass has to change. Can you please explain?

Reply:
`Rack::Session::Abstract::PersistedSecure` uses `Rack::Session::Abstract::PersistedSecure::SecureSessionHash` instead of `Rack::Session::Abstract::SessionHash`. Therefore, I changed the super class of TestSession to use SecureSessionHash as well.
The difference between SessionHash and SecureSessionHash is the behavior when accessing `["session_id"]`. In the case of SecureSessionHash, the public_id of `@id` is returned.

https://github.com/rack/rack/blob/7fecaee81f59926b6e1913511c90650e76673b38/lib/rack/session/abstract/id.rb#L461-L463